dglc/dglc_datamode_noevolve_mod.F90

```diff
@@ -380,25 +391,95 @@ subroutine dglc_datamode_noevolve_advance(pio_subsystem, io_type, io_format, &
       end if
 
       if (initialized_noevolve) then
+
          ! Compute Fgrg_rofi
          do ns = 1,num_icesheets
+
+            ! Get mesh info
             call ESMF_MeshGet(model_meshes(ns), elementdistGrid=distGrid, rc=rc)
             if (ChkErr(rc,__LINE__,u_FILE_u)) return
-            call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
-            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            lsize = size(Fgrg_rofi(ns)%ptr)
+            
+            allocate(ice_runoff_out(lsize))
+            ice_runoff_out(:) = 0.d0
+            loc_pos_smb(1) = 0.d0
+            Tot_pos_smb(1) = 0.d0
+            loc_neg_smb(1) = 0.d0
+            Tot_neg_smb(1) = 0.d0
+            num_Tot = 0.d0
+            rat = 0.d0
+
+            ! For No Evolve to reduce negative ice fluxes from DGLC, we will
+            ! Calculate the total positive and total negative fluxes on each
+            ! processor first (local totals).
             do ng = 1,lsize
-               if (Sg_icemask_coupled_fluxes(ns)%ptr(ng) > 0.d0) then
-                  Fgrg_rofi(ns)%ptr(ng) = Flgl_qice(ns)%ptr(ng)
-               else
-                  Fgrg_rofi(ns)%ptr(ng) = 0._r8
+               if (Sg_icemask_coupled_fluxes(ns)%ptr(ng).gt.0.d0) then
```

Comment:
Minor stylistic nit-pick: I feel like mathematical symbols (`>`) are easier to read than `.gt.`, and would be consistent with the other code in this module. (Here and in two other places.)

Reply:
I do too. Not sure why I did it this way. Some kind of weird code channelling old spirits of FORTRAN.